content/docs/examples/creds-in-ocmconfig.md

```diff
@@ -0,0 +1,225 @@
+
+For comprehensive documentation on the credentials topic, including usage of certificates or HashiCorp Vault, execute the command `ocm credential-handling`.
+
+### Repositories and Consumers
+
+In the examples below some configuration is located under `configurations[0]`.`repositories`, and some other under `configurations[0]`.`consumers`. This chanpter explains the difference between `repositories` and `consumers`, which is potentially not as obvious as one could think.
+
+In this context, `repository` is a place, where credentials can be stored, i.e. it is a credentials repository. For example, Docker's `config.json` can store multiple credentials, and in that sense the file serves as a repository that can store and provide credentials. That is why its location is configured under `repositories`. Other examples of credentials repositories can be the NPM's `.npmrc` file or a HashiCorp Vault instance.
+
+A `consumer` is something the credentials are required for. For example, if you need to configure credentials that are required to log in to an OCI registry, one could say that the registry will be consuming these credentials, i.e. the registry is a credentials consumer. That is why it is configured under `consumers`.
+
+### Re-use credentials configured for Docker
+
+This `.ocmconfig` file will tell OCM CLI to use credentials configuration from Docker's `config.json` file.
+
+```yaml
+type: generic.config.ocm.software/v1
+configurations:
+  - type: credentials.config.ocm.software
+    repositories:
+      - repository:
+          type: DockerConfig/v1
+          dockerConfigFile: "~/.docker/config.json"
+          propagateConsumerIdentity: true
+```
+
+### Re-use credentials configured for npm
+
+This `.ocmconfig` file will tell OCM CLI to use credentials configuration from npm's `.npmrc` file.
+
+```yaml
+type: generic.config.ocm.software/v1
+configurations:
+  - type: credentials.config.ocm.software
+    repositories:
+      - repository:
+          type: NPMConfig/v1
+          npmrcFile: '~/.npmrc'
+```
+
+### Accessing OCI registries
+
+#### HTTPS and path
+
+To access artifacts in `https://ghcr.io/open-component-model`:
+* Omit the protocol
+* The path behind the top-level domain hast to be specified in a separate `pathprefix` field
+* The `password` is the user's basic authentication password. Some OCI registries allow to generate user access tokens, which can also be used for basic authentication.
+
+```yaml
+type: generic.config.ocm.software/v1
+configurations:
+  - type: credentials.config.ocm.software
+    consumers:
+      - identity:
+          type: OCIRegistry
+          hostname: ghcr.io
+          pathprefix: open-component-model
+        credentials:
+          - type: Credentials
+            properties:
+              username: some-user
+              password: some-token
+```
+
+#### HTTP, port number, empty path
+
+To access artifacts in `http://127.0.0.1:5001`:
+* Omit the protocol

```

Comment:
Essentially the same critique as above. What's the rationale behind omitting it here?
What if the goes ahead, starts up another oci registry on port 5000, resulting in the address `http://127.0.0.1:5000`?

Reply:
What about the following?

#### HTTP, port number, empty path

To access artifacts in `http://127.0.0.1:5001`:
* The field `port` is optional. If not specified, the OCM CLI will use the credentials for all ports on that host
* As the URL has no path behind the port number, `pathprefix` element can be removed or left empty

```yaml
type: generic.config.ocm.software/v1
configurations:
  - type: credentials.config.ocm.software
    consumers:
      - identity:
          type: OCIRegistry
          scheme: http
          hostname: 127.0.0.1
          port: 5001
        credentials:
          - type: Credentials
            properties:
              username: admin
              password: admin
```

Reply:
Suggestion implemented